OsmAnd/src/net/osmand/plus/views/layers/GPXLayer.java

```diff
@@ -1237,6 +1237,7 @@ private void drawSelectedFileSegments(SelectedGpxFile selectedGpxFile, boolean c
 					geometryWay.vectorLineArrowsProvider = renderableSegment.getGeometryWay().vectorLineArrowsProvider;
 					geometryWay.updateCustomWidth(renderableSegment.getGeometryWay().getCustomWidth());
 					geometryWay.updateDrawDirectionArrows(renderableSegment.getGeometryWay().getDrawDirectionArrows());
+					resetSymbolProviders(oldSegments.get(0));
```

Comment:
Looks wrong. It is related to current recording track.

Reply:
Improved check for CurrentTrack instead of RenderableSegment should fix the issue. It happened due to calling oldSegments.remove(0); for StandardTracks